src/shared/merge.js

```diff
@@ -20,7 +20,10 @@ export function arrayMerge({ component, tagIDKeyName, metaTemplateKeyName, conte
 
     // source doesnt contain any duplicate id's
     // or the source item should be ignored
-    if (sourceIndex === -1 || sourceItem[contentKeyName] === false || sourceItem.innerHTML === false) {
+    if (sourceIndex === -1 ||
+      (sourceItem.hasOwnProperty(contentKeyName) && sourceItem[contentKeyName] === undefined) ||
+      (sourceItem.hasOwnProperty('innerHTML') && sourceItem.innerHTML === undefined)
+    ) {
```

Comment:
Just an idea: extract the conditions to well-named `const`s might help understanding the code quicker

Reply:
I have split that if statement in the 2 separate cases, together with the comment I think that should work at least as good

btw, I've added some more tests (not the e2e, those dont count for the coverage) and fixed some others. And then I fixed the issues which those tests unearthed but now we should have 100% line coverage (and damn, it was a good reminder that creating good working tests can be quite hard even though test quality could probably still be improved by a lot)